### PR TITLE
Add Multi position support and basic test infra

### PR DESCRIPTION
**Pull request by yuvals2146**

### CODE REVIEW

File: getPostionData.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ const arbitProvider = new ethers.providers.JsonRpcProvider(
 const etherProvider = new ethers.providers.JsonRpcProvider(
   process.env.ETHER_RPC_URL
 );
-const provider = process.env.CHAIN_ID === "1" ? etherProvider : arbitProvider;
+let provider;
```
Comment (azrield) [inline getPostionData.js:52]: no need for global var

File: getPostionData.js [hunk 2]
```diff
@@ -439,26 +452,12 @@ const loadPositionInitDataByTxHash = async (txhash, position) => {
 };
 
 const fixSymbolBinanceConvention = (token0symbol, token1symbol) => {
```
Comment (azrield) [inline getPostionData.js:454]: move to Binance.js

File: init.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+const { loadAllPositions } = require("./loadPositionData.js");
+const { notify } = require("./notifer.js");
+const logger = require("./logger.js");
+const { userSaveNewPosition } = require("./savePositionData.js");
+
+const init = async () => {
+  //await saveOrValidateInitPositionInfo(position);
+  const positions = (await loadAllPositions()).map((position) => {
+    if (position.hasHistoricalData) return;
```
Comment (azrield) [inline init.js:9]: don't save into DB positions without historical data

File: init.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      chain: position.chainId,
+    };
+  });
+
+  if (positions.length === 0) {
+    logger.error("No positions found in DB");
+    process.exit(0);
+  }
+
+  logger.info(`run ${positions.length} positions`, positions);
```
Comment (azrield) [inline init.js:21]: run=found

File: init.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  logger.info("init", "done");
+
+  await notify(
+    `unihedge Bot is up and running for ${positions.length} positions`,
```
Comment (azrield) [inline init.js:25]: unihedge Bot=UniApp
for=on

File: queryTheGraph.js [hunk 6]
```diff
@@ -61,7 +71,9 @@ const queryTheGraphForMintTransactHash = async (position) => {
       result.data.data.position.transaction.mints[0].id.split("#")[0];
     return txHash;
   } catch (err) {
-    logger.error("error in graph - get mint TX", err);
+    throw new Error(
+      `theGraph - could not get mint TX for position ${position.id} on chain id ${position.chain}`
+    );
```
Comment (azrield) [inline queryTheGraph.js:76]: replace chain_id with chain_name

File: savePositionData.js [hunk 7]
```diff
@@ -6,8 +6,8 @@ const { fetchHistoricalPriceData } = require("./binance.js");
 const { notify } = require("./notifer.js");
 const { chains } = require("./chains.js");
 
+const prisma = new PrismaClient();
 async function saveOrValidateInitPositionInfo(position) {
```
Comment (azrield) [inline savePositionData.js:10]: rename:
saveOrValidateInitPositionInfo => loadInitPositionInfo

File: savePositionData.js [hunk 8]
```diff
@@ -44,6 +44,10 @@ async function saveOrValidateInitPositionInfo(position) {
     return;
   }
 
+  //logger.note("Saved initial position data to SQL", position.id);
+}
+
+const saveNewPositionInDB = async (position, initData, historicalData) => {
```
Comment (azrield) [inline savePositionData.js:50]: do not save position without its historical data

File: savePositionData.js [hunk 9]
```diff
@@ -44,6 +44,10 @@ async function saveOrValidateInitPositionInfo(position) {
...
   await prisma.Position.create({
     data: {
       id: parseInt(position.id),
```
Comment (azrield) [inline savePositionData.js:53]: save chainID right after position ID

**Review by azrield — APPROVED** [review]

File: getPostionData.js (hunk 1 above)
Comment (yuvals2146) [reply getPostionData.js:52]: Done

File: analyzeRoutine.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    const currentBlockNumber = await getCurrentBlockNumber(position.chain);
+
+    analyzeDataPoint(
+      postionDataFromContract,
+      Token0USDRate,
+      Token1USDRate,
+      parseInt(position.id)
+    );
+
+    await savePositionDataSQL(
```
[the remark this answers is not in the document]
Comment (yuvals2146) [reply analyzeRoutine.js:33]: Done

File: getPostionData.js (hunk 2 above)
Comment (yuvals2146) [reply getPostionData.js:454]: Done

Comment (yuvals2146) [reply getPostionData.js:454]: rename and didnt move it

File: init.js (hunk 3 above)
Comment (yuvals2146) [reply init.js:9]: done

File: init.js (hunk 4 above)
Comment (yuvals2146) [reply init.js:21]: Done

File: init.js (hunk 5 above)
Comment (yuvals2146) [reply init.js:25]: Done

File: queryTheGraph.js (hunk 6 above)
Comment (yuvals2146) [reply queryTheGraph.js:76]: donne

File: savePositionData.js (hunk 7 above)
Comment (yuvals2146) [reply savePositionData.js:10]: removed

File: savePositionData.js (hunk 8 above)
Comment (yuvals2146) [reply savePositionData.js:50]: removed

**Pull request merged**